File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.py
class Solution:
    def fn(self, s: str, c: str) -> str:
        s1 = ""
        s2 = ""
        mp = [0] * 26
        for i in range(len(s)):
            if c == s[i] and mp[ord(s[i]) - ord('a')] == 0:
                s1 = s[:i + 1]
                s2 = s[i + 1:]
                mp[ord(s[i]) - ord('a')] = 1
        s1 = s1[::-1]
        return s1 + s2
            
    def reversePrefix(self, word: str, ch: str) -> str:
        if ch in word:
            return self.fn(word, ch)
        return word

File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.java
class Solution {
    String fn(String s , char c){
        int[] mp = new int[26];
        String s1 = "";
        String s2 = "";
        for (int i =0;i<s.length();i++){
            mp[s.charAt(i)-'a']++;
            if(c==s.charAt(i) && mp[c-'a']==1){
                s1=s.substring(0,i+1);
                s2=s.substring(i+1,s.length());
            }    
        }
        StringBuilder ans = new StringBuilder(s1);
        return ans.reverse().toString()+s2;
    }
    public String reversePrefix(String word, char ch) {
       
        if( word.contains(String.valueOf(ch))){
            return fn(word,ch);

        }
        return word;
        
    }
}